### PR TITLE
Packages update

### PR DESCRIPTION
**Pull request by Demonkratiy**

### CODE REVIEW

File: src/baseBehavior.ts [hunk 1]
```diff
@@ -53,8 +50,8 @@ export class BaseBehavior<SelectableDataPointType extends BaseDataPoint> impleme
             elementsSelection
         } = this.options;
 
-        elementsSelection.on("click", (datum) => {
-            const mouseEvent: MouseEvent = <MouseEvent>getEvent() || <MouseEvent>window.event;
+        elementsSelection.on("click", (event, datum) => {
+            const mouseEvent: MouseEvent = <MouseEvent>event || <MouseEvent>window.event;
```
Comment (uve) [inline src/baseBehavior.ts:54]: run d3 select for elementsSelection as they come from the visual.

File: src/baseBehavior.ts [hunk 2]
```diff
@@ -66,8 +63,8 @@ export class BaseBehavior<SelectableDataPointType extends BaseDataPoint> impleme
         const {
             clearCatcherSelection
         } = this.options;
-        clearCatcherSelection.on("click", () => {
-            const mouseEvent: MouseEvent = <MouseEvent>getEvent() || <MouseEvent>window.event;
+        clearCatcherSelection.on("click", (event) => {
```
Comment (uve) [inline src/baseBehavior.ts:66]: the same for clearCatcher

File: src/baseBehavior.ts [hunk 3]
```diff
@@ -66,8 +63,8 @@ export class BaseBehavior<SelectableDataPointType extends BaseDataPoint> impleme
...
+            const mouseEvent: MouseEvent = <MouseEvent>event || <MouseEvent>window.event;
```
Comment (uve) [inline src/baseBehavior.ts:67]: we can ignore window.event here. d3-select will do everything for us

File: src/baseBehavior.ts [hunk 4]
```diff
@@ -108,20 +105,20 @@ export class BaseBehavior<SelectableDataPointType extends BaseDataPoint> impleme
             }
         };
 
-        clearCatcherSelection.on("contextmenu", () => {
-            const event: MouseEvent = <MouseEvent>getEvent() || <MouseEvent>window.event;
-            if (event) {
+        clearCatcherSelection.on("contextmenu", (event) => {
+            const e: MouseEvent = <MouseEvent>event || <MouseEvent>window.event;
```
Comment (uve) [inline src/baseBehavior.ts:109]: we can ignore window.event here. d3-select will do everything for us

File: src/interactivityBaseService.ts [hunk 5]
```diff
@@ -97,7 +99,7 @@ export interface ISelectionHandler {
      * identity is undefined, the selection state is cleared. In this case, if specificIdentity
      * exists, it will still be sent to the host.
      */
-    handleSelection(dataPoints: BaseDataPoint | BaseDataPoint[], multiSelect: boolean): void;
+    handleSelection(dataPoints: BaseDataPoint | BaseDataPoint[], multiSelect: boolean, event?): void;
```
Comment (uve) [inline src/interactivityBaseService.ts:102]: why do you create additional argument here?

File: src/interactivityUtils.ts [hunk 6]
```diff
@@ -25,36 +25,29 @@
...
 }
-
-export function registerStandardSelectionHandler(selection: d3.Selection<any, any, any, any>, selectionHandler: ISelectionHandler): void {
-    selection.on("click", (d: SelectableDataPoint) => handleSelection(d, selectionHandler));
+export function registerStandardSelectionHandler(selection: Selection<any, any, any, any>, selectionHandler: ISelectionHandler): void {
+    selection.on("click", (event, d: SelectableDataPoint) => handleSelection(d, selectionHandler, event));
 }
+export function registerGroupSelectionHandler(group: Selection<any, any, any, any>, selectionHandler: ISelectionHandler): void {
+    group.on("click", (event) => {
+        let target: EventTarget = (<MouseEvent>event).target,
```
Comment (uve) [inline src/interactivityUtils.ts:44]: run it from the visual to verify.

**Review by uve — CHANGES_REQUESTED** [review]

File: src/baseBehavior.ts (hunk 1 above)
Comment (Demonkratiy) [reply src/baseBehavior.ts]: done

File: src/baseBehavior.ts (hunk 2 above)
Comment (Demonkratiy) [reply src/baseBehavior.ts]: done

File: src/baseBehavior.ts (hunk 3 above)
Comment (Demonkratiy) [reply src/baseBehavior.ts]: window.event removed

File: src/baseBehavior.ts (hunk 4 above)
Comment (Demonkratiy) [reply src/baseBehavior.ts]: window.event removed

File: src/interactivityBaseService.ts (hunk 5 above)
Comment (Demonkratiy) [reply src/interactivityBaseService.ts]: Additional argument removed.

File: src/interactivityUtils.ts (hunk 6 above)
Comment (Demonkratiy) [reply src/interactivityUtils.ts]: Tested on Mekko with using of build "lib" inside of visual's node_modules

File: src/baseBehavior.ts [hunk 7]
```diff
@@ -82,17 +84,19 @@ export class BaseBehavior<SelectableDataPointType extends BaseDataPoint> impleme
             elementsSelection
         } = this.options;
 
-        elementsSelection.on("contextmenu", (datum) => {
-            const event: MouseEvent = <MouseEvent>getEvent() || <MouseEvent>window.event;
-            if (event) {
+        let internalSelection: Selection<any, SelectableDataPoint, any, any> = selectAll(elementsSelection.nodes());
+
+        internalSelection.on("contextmenu", (event, datum) => {
+            const e: MouseEvent = <MouseEvent>event;
```
Comment (uve) [inline src/baseBehavior.ts:90]: Why do you create a new variable here?

File: src/baseBehavior.ts [hunk 8]
```diff
@@ -108,20 +112,22 @@ export class BaseBehavior<SelectableDataPointType extends BaseDataPoint> impleme
             }
         };
 
-        clearCatcherSelection.on("contextmenu", () => {
-            const event: MouseEvent = <MouseEvent>getEvent() || <MouseEvent>window.event;
-            if (event) {
+        let internalSelection: Selection<any, SelectableDataPoint, any, any> = selectAll(clearCatcherSelection.nodes());
+        
+        internalSelection.on("contextmenu", (event) => {
+            const e: MouseEvent = <MouseEvent>event;
```
Comment (uve) [inline src/baseBehavior.ts:118]: Why do you create a new variable here?

**Review by uve — CHANGES_REQUESTED** [review]

File: src/baseBehavior.ts (hunk 7 above)
Comment (Demonkratiy) [reply src/baseBehavior.ts]: It was used in previous version with d3.v5 getEvent method. 
So I forgot to refactor this part, you are right, now as we receive native event, this variable is redundant and removed in my last commit.

File: src/baseBehavior.ts (hunk 8 above)
Comment (Demonkratiy) [reply src/baseBehavior.ts]: It was used in previous version with d3.v5 getEvent method. 
So I forgot to refactor this part, you are right, now as we receive native event, this variable is redundant and removed in my last commit.

**Review by uve — APPROVED** [review]

**Pull request merged**